### PR TITLE
Dsnd 363 retry error kafka mechanism

### PR DESCRIPTION
**Pull request by amobarak**
Retry error kafka mechanism, the config is as follows:
image

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/insolvency/delta/consumer/InsolvencyDeltaConsumer.java [hunk 1]
```diff
@@ -14,36 +23,39 @@
...
-        deltaProcessor.processDelta(chsDeltaMessage);
-    }
-
-    /**
-     * Receives Retry topic messages.
-     */
-    @KafkaListener(id = "${insolvency.delta.retry-id}",
-            topics = "${insolvency.delta.topic.retry}",
+    @RetryableTopic(attempts = "4",
+            backoff = @Backoff(delay = 100),
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/insolvency/delta/consumer/InsolvencyDeltaConsumer.java:43]: Attempts and backoff period should be configurable in apllication yml file

File: src/main/java/uk/gov/companieshouse/insolvency/delta/serialization/ChsDeltaSerializer.java [hunk 2]
```diff
@@ -1,28 +1,57 @@
...
+
         try {
-            DatumWriter<ChsDelta> writer = new SpecificDatumWriter<>();
-            EncoderFactory encoderFactory = EncoderFactory.get();
+            if (payload == null) {
+                return null;
+            }
+
+            if (payload instanceof byte[]) {
+                return (byte[]) payload;
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/insolvency/delta/serialization/ChsDeltaSerializer.java:37]: Probably better to write tests for different conditional blocks on serializer and de-serialzer

File: src/main/resources/chs-delta.avsc [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+  "type": "record",
+  "namespace": "uk.gov.companieshouse.delta",
+  "name": "ChsDelta",
+  "doc": "Schema for the delta that will be used to transfer data from CHIPS to CHS.",
+  "fields": [
+    {
```
Comment (zahidm3) [inline src/main/resources/chs-delta.avsc:7]: We can delete this, sorry, I added for some testing

**Review by gseregni-ch — APPROVED** [review]
Please resolve the conflicts before merge.

**Pull request merged**